c_sharp_json_diff/DiffProcessor.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace c_sharp_json_diff
+{
+    abstract class DiffProcessor : IJsonProcessor<object>
+    {
+        private const string SkippableProperty = "_t";
+        private const string propertyLeft = "left";
+        private const string propertyRight = "right";
+
+        public void Process(string key, JObject jObject)
+        {
+            foreach (KeyValuePair<string, JToken?> keyValuePair in jObject)
+            {
+                if (ShouldSkipProperty(keyValuePair.Key))
+                {
+                    continue;
+                }
+
+                if (keyValuePair.Value != null)
+                {
+                    if (keyValuePair.Value is JArray valuesAsArray)
+                    {
+                        if (valuesAsArray.Count == 2)
+                        {
+                            if (valuesAsArray[0] != null && valuesAsArray[1] != null)
+                            {
+                                Dictionary<string, object> newJsonObj =  PerformDiffProcess(valuesAsArray[0], valuesAsArray[1]);
+                                Dictionary<object, object> dict = new Dictionary<object, object>();
+                                if (newJsonObj != null && newJsonObj.Count != 0) 
+                                {
+                                    dict = CreateDictionary(key, valuesAsArray[0], valuesAsArray[1], newJsonObj);
+                                    keyValuePair.Value.Replace(JToken.FromObject(dict));
+                                }
+                            }
+                        }
+                    }
+                    else
+                    {
+                        if(keyValuePair.Value is JObject jsonObj)
+                        {
+                            Process(key, jsonObj);
+                        }
+                    }
+                }
+            }
+        }
+
+        public abstract Dictionary<string, object> PerformDiffProcess(JToken left, JToken right);
```

Comment:
Add comment-documentation here about what this dictionary represents. Include in the documentation that an empty or null dictionary means we skip the diff.